charts/xrd-vrouter/templates/statefulset.yaml

```diff
@@ -55,14 +55,32 @@ Construct the resources including the default if that resource wasn't specified.
 
 {{- /* Generate CPU env vars */}}
 {{- if .Values.cpu }}
-  {{- with .Values.cpu.cpuset }}
-    {{- $_ := set $env "XR_VROUTER_CPUSET" . }}
-  {{- end }}
-  {{- with .Values.cpu.controlPlaneCpuCount }}
-    {{- $_ := set $env "XR_VROUTER_CP_NUM_CPUS" . }}
-  {{- end }}
-  {{- with .Values.cpu.hyperThreadingMode }}
-    {{- $_ := set $env "XR_VROUTER_HT_MODE" . }}
+  {{- with .Values.cpu}}
+    {{- if and .controlPlaneCpuset (not .dataPlaneCpuset)}}
+      {{- fail "controlPlaneCpuset is set but dataPlaneCpuset is not set" }}
```

Comment:
Not sure if these error messages are deliberately along the same lines of what we already have, but if not, then I'd prefer if these were statements of "what is wrong and how to fix it", not describing "what is happening".  This generally makes it easier for the end-user to immediately understand the problem and fix it:

- "dataPlaneCpuset must be set if controlPlaneCpuset is set"
- "controlPlaneCpuset must be set if dataPlaneCpuset is set"
- "cpuset must not be set if controlPlaneCpuset and dataPlaneCpuset are set"
- "controlPlaneCpuCount must not be set if controlPlaneCpuset and dataPlaneCpuset are set"

There's also an argument that these should all be prefixed with `cpu.` - do you have a preference?  I probably lean towards no - it's clear enough that these are CPU settings that live under `cpu`, and adding the prefix would just add noise.

Reply:
Implemented the suggested error messages. RE `cpu.`: other error messages, e.g. for `XR_INTERFACES` env var in `charts/xrd-vrouter/templates/_interfaces.tpl` do not include the prefix and I don't think we should here either.